platform.md

```diff
@@ -396,16 +339,69 @@ When analyzing a given layer the lifecycle SHALL:
     - Write layer metadata read from the cache to `<layers>/<buildpack-id>/<layer-name>.toml`
     - Write the layer diffID from the cache to `<layers>/<buildpack-id>/<layer-name>.sha`
 
+#### `detector`
+The platform MUST execute `detector` in the **build environment**
+
+Usage: 
+```
+/cnb/lifecycle/detector \
+  [-app <app>] \
+  [-buildpacks <buildpacks>] \
+  [-group <group>] \
+  [-layers <layers>] \
+  [-log-level <log-level>] \
+  [-order <order>] \
+  [-plan <plan>] \
+  [-platform <platform>]
+```
+
+##### Inputs
+| Input         | Environment Variable    | Default Value             | Description
+|---------------|-------------------------|---------------------------|----------------------
+| `<app>`         | `CNB_APP_DIR`         | `/workspace`          | Path to application directory
+| `<buildpacks>`  | `CNB_BUILDPACKS_DIR`  | `/cnb/buildpacks`     | Path to buildpacks directory (see [Buildpacks Directory Layout](#buildpacks-directory-layout))
+| `<group>`       | `CNB_GROUP_PATH`      | `<layers>/group.toml` | Path to output group definition
+| `<layers>`      | `CNB_LAYERS_DIR`      | `/layers`             | Path to layers directory
+| `<log-level>`   | `CNB_LOG_LEVEL`       | `info`                | Log Level
+| `<order>`       | `CNB_ORDER_PATH`      | `/cnb/order.toml`     | Path to order definition (see [`order.toml`](#ordertoml-toml))
+| `<plan>`        | `CNB_PLAN_PATH`       | `<layers>/plan.toml`  | Path to output resolved build plan
+| `<platform>`    | `CNB_PLATFORM_DIR`    | `/platform`           | Path to platform directory
+
+##### Outputs
+| Output             | Description
+|--------------------|----------------------------------------------
+| [exit status]      | (see Exit Code table below for values)
+| `/dev/stdout`      | Logs (info)
+| `/dev/stderr`      | Logs (warnings, errors)
+| `<group>`          | Detected buildpack group  (see [`group.toml`](#grouptoml-toml))
+| `<plan>`           | Resolved Build Plan (see [`plan.toml`](#plantoml-toml))
+
+| Exit Code | Result|
+|-----------|-------|
+| `0`       | Success
+| `11`      | Platform API incompatibility error
+| `12`      | Buildpack API incompatibility error
+| `1-10`, `13-99` | Generic lifecycle errors
+| `100`     | All buildpacks groups have failed to detect w/o error
+| `101`     | All buildpack groups have failed to detect and at least one buildpack has errored
+| `102-199` | Detection-specific lifecycle errors
+
+The lifecycle:
+- SHALL detect a single group from `<order>` and write it to `<group>` using the [detection process](buildpack.md#phase-1-detection) outlined in the Buildpack Interface Specification
+- SHALL write the resolved build plan from the detected group to `<plan>`
+
 #### `restorer`
 Usage:
 ```
 /cnb/lifecycle/restorer \
-  [-cache-dir <cache-dir>]
-  [-cache-image <cache-image>]
+  [-cache-dir <cache-dir>] \
+  [-cache-image <cache-image>] \
+  [-daemon] \ # sets <daemon> 
```

Comment:
I think this gets confusing in conjunction with our registry only cache image. Maybe we can use the analysis metadata to record whether the previous image is a daemon or registry image?

Also, this might be a really good opportunity to make a long desired change and move from have a single `reference` key in [analyzed.toml](https://github.com/buildpacks/spec/blob/main/platform.md#analyzedtoml-toml) that stores either an imageID or digest reference, to having two explicit keys (`image_id`, `registry_digest`)

Reply:
Apologies I thought I updated this comment with an edit. I realized existing `analyzer` already has this issue (if you want to call it an issue) and this suggestion should probably be out of scope for now. But I wanted to leave these thoughts somewhere, in case we want to incorporate them in future changes.